backend/lib/processor.py

```diff
@@ -418,7 +418,7 @@ def add_field_to_parent(self, field_name, new_data, which_parent=source_dataset,
 		parent_path = which_parent.get_results_path()
 
 		if len(new_data) != which_parent.num_rows:
-			raise ProcessorException('Must have new data point for each record: parent dataset: %i, new data points: %i' % (which_parent.num_rows, len(new_data)))
+			self.dataset.update_status('The amount of new data points and existing records don\'t match; data may be misaligned (parent dataset: %i, new data points: %i)' % (which_parent.num_rows, len(new_data)))
```

Comment:
The issue was that there is no check ensuring new data is added to the correct row (for example, a check that the `id` is the same). If a row is missing, the function will naively add the data to whatever new row is at that _index_. It was raising an error to avoid adding the data to the wrong row/item.

Reply:
Since this method is now no longer used for annotations or the explorer as far as I can see, but only for other processors, I would revert the changes made to it